kafka-connect-s3/src/main/java/io/confluent/connect/s3/hooks/BlockingKafkaPostCommitHook.java

```diff
@@ -0,0 +1,146 @@
+      log.error("Failed to begin transaction with unrecoverable exception, closing producer", e);
+      kafkaProducer.close();
+      throw new ConnectException(e);
+    } catch (KafkaException e) {
+      log.error("Failed to begin transaction", e);
+      throw new RetriableException(e);
+    }
+  }
+
+  private void sendRecords(Set<String> s3ObjectPaths) {
+    try {
+      for (String s3ObjectPath : s3ObjectPaths) {
+        kafkaProducer.send(new ProducerRecord<>(kafkaTopic, s3ObjectPath));
+      }
+    } catch (IllegalWorkerStateException e) {
+      log.error("Failed to send record because no transaction.id or "
+              + "initTransaction() wasn't called", e);
+      kafkaProducer.close();
+      throw new ConnectException(e);
+    } catch (ProducerFencedException | OutOfOrderSequenceException | AuthorizationException
+             | UnsupportedVersionException e) {
+      log.error("Failed to send with unrecoverable exception, closing producer", e);
+      kafkaProducer.close();
+      throw new ConnectException(e);
+    } catch (KafkaException e) {
+      log.error("Failed to send aborting transaction, going to retry later", e);
+      kafkaProducer.abortTransaction();
+      throw new RetriableException(e);
+    }
+  }
+
+  private void commitTransaction() {
+    try {
+      kafkaProducer.commitTransaction();
+      log.info("committed transaction successfully");
+    } catch (ProducerFencedException | AuthorizationException | UnsupportedVersionException
+             | IllegalWorkerStateException e) {
+      log.error("Failed to commit transaction with unrecoverable exception, "
+              + "closing producer", e);
+      kafkaProducer.close();
+      throw new ConnectException(e);
+    } catch (TimeoutException | InterruptException e) {
+      log.error("Failed to commit transaction due to timeout or interrupt", e);
+      kafkaProducer.close();
+      throw new ConnectException(e);
+    } catch (KafkaException e) {
+      log.error("aborting transaction, going to retry later", e);
+      kafkaProducer.abortTransaction();
+      throw new RetriableException(e);
+    }
+  }
+
+  @Override
+  public void close() {
+    if (kafkaProducer != null) {
+      try {
+        kafkaProducer.close();
+      } catch (Exception e) {
+        log.error("Failed to close kafka producer", e);
+      }
+      kafkaProducer = null;
+    }
+  }
+
+  private KafkaProducer<String, String> newKafkaPostCommitProducer(S3SinkConnectorConfig config) {
+    Properties props = new Properties();
+    props.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG,
+            config.getPostCommitKafkaBootstrapBrokers());
+    props.setProperty(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+            StringSerializer.class.getName());
+    props.setProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+    props.setProperty(ProducerConfig.CLIENT_ID_CONFIG, "blocking-kafka-producer-"
```

Comment:
I agree. I didn't find a way to get taskId from API. 
Another workaround for this will be to print a log binding the set of topic partition and their hash code.